<?xml version="1.0" encoding="UTF-8"?>
<launch>




  <arg name="model_path" value="/mnt/VIJAY/thesis/Model_Weights/Monitoring_System/openpcdet/pv_rcnn.pth"/>
  <arg name="cfg_path" value="/3D-object-detection/OpenPCDet/tools/cfgs/kitti_models/pv_rcnn.yaml"/>
  <arg name="input_pointcloud_topic" value="/os1_cloud_node/points_sync"/>
  <arg name="translate_x" value= "25"/>
  <arg name="threshold_score" value= "0.25"/>


  <node pkg="deploy" type="openpcdet_node.py" name="openpcdet_node" output="screen" required="true">
    <param name="model_path" value="$(arg model_path)"/>
    <param name="cfg_file" value="$(arg cfg_path)"/>
    <param name="input_pointcloud_topic" value="$(arg input_pointcloud_topic)"/>
    <param name="translate_x" value="$(arg translate_x)"/>
    <param name="threshold_score" value="$(arg threshold_score)"/>

    <remap from="check_for_3D_objects" to="/open_pcdet/check_for_3D_objects"/>

  </node>

  <include file="$(find deploy)/launch/play_rosbag.launch" />
  <!-- <arg name="topics_name" default="/heatmapper3d_efficientdetrgb/layered_heatmaps
                                  /heatmapper3d_efficientdetrgb/occupancy_cloud
                                  /heatmapper3d_efficientdetthermal/layered_heatmaps
                                  /heatmapper3d_efficientdetthermal/occupancy_cloud"/> -->
  <!-- <arg name="topics_name" default="/heatmapper3d_3ddetection/layered_heatmaps
                                  /heatmapper3d_3ddetection/occupancy_cloud
                                  "/>
    <node pkg="rosbag" type="record" name="rosbag_record" output="screen" args="$(arg topics_name) -O /mnt/VIJAY/thesis/rosbags/wms_thesis_new_threed/crowded/openpcdet_PartA2_free.bag ">
    </node> -->


</launch>

              
